docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py

```diff
@@ -64,36 +75,38 @@ def test_times(helpers):
 @pytest.mark.strings
 def test_letter(helpers):
     helpers.set_student_file('letter.py')
-    helpers.expected_pycode = [
-       [r'''\b[print(f]['|"][{].*[date].*[}\\n\\tFor\\n\\t{].*[name].*[}].*''',
-        'make sure to use \\t, \\n and f\' in your print statement.'],
-    ]
-    helpers.exact_answer = True
-    expected_msg = ('{}\n\tFor\n\t{}\n\t{}\n' +
-                    '\nDear Mr./Mrs. {}\n' +
-                    'Please visit our office as soon as possible to arrange ' +
-                    'your payments.\n' +
-                    'We can\'t wait until it’s all done ...\n\n' +
-                    'Sincerely\n\tKoogle Inc.\n\t{}')
     helpers.input = [
         ['3.3.20', 'Anna', 'Rasbery, US', 'Hotel california, US'],
-        ['5.4.21', 'Yael', 'Herzel 5, Haifa', 'Yam 1, Tel Aviv']
     ]
-    helpers.expected_stdout = [
-        [r'.*' + expected_msg.format(helpers.input[0][0],
-                                     helpers.input[0][1],
-                                     helpers.input[0][2],
-                                     helpers.input[0][1],
-                                     helpers.input[0][3]),
-         f'For the input: {helpers.input[0]},' +
-         'some of the expected text was missing'],
-        [r'.*' + expected_msg.format(helpers.input[1][0],
-                                     helpers.input[1][1],
-                                     helpers.input[1][2],
-                                     helpers.input[1][1],
-                                     helpers.input[1][3]),
-         f'For the input: {helpers.input[0]},' +
-         'some of the expected text was missing']
+    error_message_for_partial_letter = (
+        f'For the input: {helpers.input[0]},' +
+        'some of the expected text was missing'
+        )
+
+    helpers.tests_list = [
+        [r'''\bprint\(f['|"]\{.*[date].*\}\\n\\tFor\\n\\t\{.*[name].*\}.*''',
+         'make sure to use \\t, \\n and f\' in your print statement.'],
```

Comment:
This one can be very tricky, as not all students know how to use tab (\t) or necessarily use newlines (\n).

Reply:
Both of those are included in the slides on page 5.4, that's why I tested for them. If you think another solution is in order, I am open for ideas.

Reply:
Maybe the test should check a regex for the tabs and new lines?
I think the text itself is more important here.

Reply:
Hi @sleviim , updated the check as discussed.
Removed the text check, only the first line is checked ans the secondary solution checks for the tabs only.